src/Network/Ethereum/Web3/Solidity/Generic.purs

```diff
@@ -215,12 +215,9 @@ dParser = do
   lookAhead
     $ do
         (ParseState _ (Position p) _) <- get
-        _ <- take (dataOffset * 2 - (p.column - 1))
+        _ <- takeBytes (dataOffset - (p.column - 1))
```

Comment:
I'm actually surprised this still works: If you look at the definition of `takeByte`, the parser position state is incremented once per byte. Yet the definition of `dParser` is using the position relative to the characters 🤔

Reply:
let's change

```
takeByte :: forall m. Monad m => ParserT HexString m HexString
takeByte = do
  ParseState input (Position position) _ <- get
  if hexLength input < 2 then
    fail "Unexpected EOF"
  else do
    let
      { after, before } = splitAt 2 (unHex input)

      unsafeMkHex s = unsafePartial $ fromJust $ mkHexString s

      position' = Position $ position { column = position.column + 1 }
    put $ ParseState (unsafeMkHex after) position' true
    pure $ unsafeMkHex before
```

to

```
takeByte :: forall m. Monad m => ParserT HexString m HexString
takeByte = do
  ParseState input (Position position) _ <- get
  if hexLength input < 2 then
    fail "Unexpected EOF"
  else do
    let
      { after, before } = splitAt 2 (unHex input)

      unsafeMkHex s = unsafePartial $ fromJust $ mkHexString s

      position' = Position $ position { column = position.column + 2 } -- !!!!
    put $ ParseState (unsafeMkHex after) position' true
    pure $ unsafeMkHex before
```

?